src/Illuminate/Database/Eloquent/Concerns/HasRelationships.php

```diff
@@ -792,17 +792,22 @@ public function touches($relation)
      */
     public function touchOwners()
     {
-        foreach ($this->getTouchedRelations() as $relation) {
-            $this->$relation()->touch();
-
-            if ($this->$relation instanceof self) {
-                $this->$relation->fireModelEvent('saved', false);
-
-                $this->$relation->touchOwners();
-            } elseif ($this->$relation instanceof Collection) {
-                $this->$relation->each->touchOwners();
-            }
-        }
+        without_recursion(
+            function () {
+                foreach ($this->getTouchedRelations() as $relation) {
+                    $this->$relation()->touch();
+
+                    if ($this->$relation instanceof self) {
+                        $this->$relation->fireModelEvent('saved', false);
+
+                        $this->$relation->touchOwners();
+                    } elseif ($this->$relation instanceof Collection) {
+                        $this->$relation->each->touchOwners();
+                    }
+                }
+            },
+            null,
```

Comment:
There isn't a default for the second parameter - my thinking was you should be explicit about what is returned if the function is called again. In situations like this where you don't want to return anything, it's currently set up to require an explicit `null`

Reply:
I think it would make sense to default the param to null like in the previous helper, this is just less verbose for void functions

Reply:
Yeah, possibly.

Mostly I was trying to force being explicit so that people don't end up with Heisenbugs because the rare occasion that they actually _encounter_ recursion returns a `null` instead of whatever is expected...